lib/fog/vsphere/requests/compute/list_vm_interfaces.rb

```diff
@@ -62,14 +62,18 @@ def get_raw_interface(vm_id, options = {})
         private
 
         # rubocop:disable Style/ConditionalAssignment
-        def raw_to_hash(nic)
+        def raw_to_hash(nic, datacenter = nil)
           if nic.backing.respond_to?(:network)
             network = nic.backing.network.name
           elsif nic.backing.respond_to?(:port)
             network = nic.backing.port.portgroupKey
-          else
-            network = nil
-          end
+          elsif nic.backing.respond_to?(:opaqueNetworkId)
+             opaquenetworks = list_networks({ :datacenter => datacenter }).select{|net| net.key?(:opaqueNetworkId)}
+  	     network = opaquenetworks.find {|opaquenetwork|  	    
```

Comment:
Multiline blocks should use `do..end` instead of `{}`

```suggestion
  	     network = opaquenetworks.find do |opaquenetwork|
```